Deprecate Spark recovery, update Flink docs (#449)
snowplow/documentation: docs/managing-data-quality/snowplow-event-recovery/running/flink/index.md

#### Running

> Event recovery jobs are usually ran using our [dataflow-runner](https://github.com/snowplow/dataflow-runner) application. An example configuration for running is [available in the repository](https://github.com/snowplow-incubator/snowplow-event-recovery/blob/master/.dataflow-runner).

Reviewer: ```suggestion
Event recovery jobs are usually run using our [dataflow-runner](https://github.com/snowplow/dataflow-runner) application. An example configuration for running is [available in the repository](https://github.com/snowplow-incubator/snowplow-event-recovery/blob/master/.dataflow-runner).
```

Author: That's passive voice, so I suppose the correct form is `ran`

To run a transient EMR cluster and execute the job, run:

```bash
dataflow-runner run-transient \

>   --emr-playbook $REPO_DIR/spark-playbook.json.tmpl \

Reviewer: What is $REPO_DIR?

  --emr-config $REPO_DIR/spark-cluster.json.tmpl \

>   --vars bucket,$BUCKET_INPUT,region,AWS_$REGION,subnet,$AWS_SUBNET,role,$AWS_IAM_ROLE,keypair,$AWS_KEYPAIR,client,$JOB_OWNER,version,$RECOVERY_VERSION,config,$RECOVERY_CONFIG,resolver,$IGLU_RESOLVER,output,$KINESIS_OUTPUT,inputdir,$BUCKET_INPUT_DIRECTORY,interval,$INTERVAL

Reviewer: ```suggestion
  --vars bucket,$BUCKET_INPUT,region,$AWS_REGION,subnet,$AWS_SUBNET,role,$AWS_IAM_ROLE,keypair,$AWS_KEYPAIR,client,$JOB_OWNER,version,$RECOVERY_VERSION,config,$RECOVERY_CONFIG,resolver,$IGLU_RESOLVER,output,$KINESIS_OUTPUT,inputdir,$BUCKET_INPUT_DIRECTORY,interval,$INTERVAL
```

Reviewer: Is there any documentation for all these vars? As a user I would have no idea what to put in many of them

#### Monitoring

EMR Flink Job emits a wide range of metrics through Cloudwatch Agent which is installed during cluster bootstrap using a provided script (see project repo's dataflow-runner directory). Custom metrics include:
- number of unrecoverable events in the run: `taskmanager_container_XXXX_Event recovery job started__Process_0_events_unrecoverable`
- number of failed recovery attempts: `taskmanager_container_XXXX_Event recovery job started__Process_0_events_failed`
- number of recovered events: `taskmanager_container_XXXX_Event recovery job started__Process_0_events_recovered`
- total number of events submitted for processing: `taskmanager_container_XXXX_Event recovery job started__Process_0_numRecordsIn`

The metrics are delivered to `snowplow/event-recovery` Cloudwatch namespace.

Given a one-second aggregation in Cloudwatch the diagram should show an always increasing metrics that should balance themselves up to total sum.

:::note

> Please note, that Flink's `numRecordsOut` metrics for sinks does not reflect an actual number of records saved in the sink.

Reviewer: ```suggestion
Flink's `numRecordsOut` metric for sinks does not reflect an actual number of records saved in the sink.
```

```bash
dataflow-runner run-transient \
  --emr-playbook flink-playbook.json.tmpl \
  --emr-config flink-cluster.json.tmpl \
  --vars bucket,$BUCKET_INPUT,region,$AWS_REGION,subnet,$AWS_SUBNET,role,$AWS_IAM_ROLE,keypair,$AWS_KEYPAIR,client,$JOB_OWNER,version,$RECOVERY_VERSION,config,$RECOVERY_CONFIG,resolver,$IGLU_RESOLVER,output,$KINESIS_OUTPUT,inputdir,$BUCKET_INPUT_DIRECTORY,interval,$INTERVAL
  ```
Where:
- `BUCKET_INPUT` - S3 bucket containing bad events to be recovered (eg. geoffs-bad-events)
- `BUCKET_INPUT_DIRECTORY` - directory in the `BUCKET_INPUT` to use as the source for bad events
- `AWS_REGION` - region in which the job is being ran (eg. eu-central-1)
- `AWS_SUBNET` - network subnet to run the job in (eg. subnet-435010347a21886ab)
- `AWS_IAM_ROLE` - AWS IAM role to assume while running the job (eg. geoffs-recovery-role)
- `AWS_KEYPAIR` - AWS EC2 key pair to use for the instances (eg. geoffs-keypair)
- `JOB_OWNER` - tag to use to mark the owner of the job (eg. goeff)
- `RECOVERY_VERSION` - application version (eg. 0.6.0)

> - `RECOVERY_CONFIG` - recovery job config as described in [Concepts](../concepts)

Reviewer: ```suggestion
- `RECOVERY_CONFIG` - recovery job config as described in [Concepts](../concepts/index.md)
```
Here and below — see README for the docs repo on why this is needed

Author: @stanch I believe this is what I had before, but this fails the build, so I turned to top-level link.

Reviewer: It fails the build because the link is wrong :) So actually the link checking feature works as expected!
For some reason, there is no `../concepts/index.md`, but there is `../0-2-concepts/index.md`.

Reviewer: https://docs.snowplow.io/docs/managing-data-quality/snowplow-event-recovery/concepts/ — this is a 404

Reviewer: Alternatively, feel free to rename that dir to `concepts` and add a redirect in `static/_redirects`
